src/utils/BlockUtils.ts

```diff
@@ -110,7 +115,11 @@ export class BlockFinder {
       if (timestamp >= block.timestamp) return block;
     }
 
-    // Check the first block. If it's grater than our timestamp, we need to find an earlier block.
+    // Prime the BlockFinder cache with any supplied hints.
+    // If the hint is accurante then this will bypass the subsequent estimation.
+    await Promise.all(Object.values(hints).map((blockNumber) => this.getBlock(blockNumber)));
```

Comment:
Hardcoding it doesn't work super well because both `lowBlock` and `highBlock` can be undefined, and this would cause an issue in `getBlock()`. `mapAsync()` seems like an OK candidate, but I've updated the code to include a filter stage to prevent supplying an undefined block number, so I don't really see much opportunity to simplify this now anyway :shrug: